File: src/web/magmac/app/stage/ParseUnit.ts
import { NodeList } from "../../../magmac/app/compile/node/NodeList";
import { ParseState } from "../../../magmac/app/stage/parse/ParseState";
import { BiFunction } from "../../../java/util/function/BiFunction";
export interface ParseUnit { toLocationUnit() : Unit<T>;<R> R merge) : merge(BiFunction<ParseState, T, R>; retainWithList() : ParseUnit<NodeList>; left() : ParseState; right() : T;
}

File: packages/compiler-java/src/main/java/magmac/app/stage/unit/ParseUnit.java
package magmac.app.stage.unit;

import magmac.app.compile.node.NodeList;
import magmac.app.stage.parse.ParseState;

import java.util.function.BiFunction;

/**
 * Compilation unit produced by the parser.
 */
public interface ParseUnit<T> {
    Unit<T> toLocationUnit();

    <R> R merge(BiFunction<ParseState, T, R> merge);

    ParseUnit<NodeList> retainWithList();

    @Deprecated
    ParseState left();

    @Deprecated
    T right();
}
